processing/src/main/java/org/apache/druid/query/aggregation/first/NumericFirstVectorAggregator.java

```diff
@@ -164,7 +215,7 @@ void updateTimeWithNull(ByteBuffer buf, int position, long time)
    * Abstract function which needs to be overridden by subclasses to set the
    * latest value in the buffer depending on the datatype
    */
-  abstract void putValue(ByteBuffer buf, int position, int index);
+  abstract void putValue(ByteBuffer buf, int position, Number number);
```

Comment:
Let's revert this change if we can. It will lead to some duplication, however, this will lead to the boxing of the primitives, which is what the column selectors aim to reduce (if they can). Therefore there are separate methods like `isNull` + `getLong` which are preferred instead of `getObject` because the former doesn't auto-box the primitive variable to an object.